fetch-in-react-practice/client/components/app.jsx

```diff
@@ -93,8 +93,9 @@ export default class App extends React.Component {
     })
       .then(response => response.json())
       .then(data => {
-        todos[index] = data;
-        this.setState({ todos });
+        const newTodos = this.state.todos;
+        newTodos[index] = data;
```

Comment:
this is actually a reference to the same array. you can use slice or spread to create the copy.